scene/gui/popup_menu.cpp

```diff
@@ -298,9 +298,16 @@ void PopupMenu::_submenu_timeout() {
 	submenu_over = -1;
 }
 
-void PopupMenu::gui_input(const Ref<InputEvent> &p_event) {
-	ERR_FAIL_COND(p_event.is_null());
+void PopupMenu::_input_from_window(const Ref<InputEvent> &p_event) {
+	if (p_event.is_valid()) {
+		_input_from_window_internal(p_event);
+	} else {
+		WARN_PRINT("Invalid event received.");
```

Comment:
Shouldn't this fail instead of just warning? You are sending the invalid event further.

Reply:
Previously, invalid events were also sent via the `window_input`-signal. My reason for this implementation was, that I didn't want to change the previous behavior.
Please let me know, if it is fine to fail on an invalid InputEvent, then I will adjust the PR.

Reply:
Well I was worried, because some classes use the event without checking if it's valid, but now I see that they are unrelated to this code path.

Still, the warning could be more descriptive I think.

Reply:
You are right. I have made the comment more explicit.